packages/discovery-react-components/src/utils/document/documentUtils.ts

```diff
@@ -99,13 +89,26 @@ export function getTextNodeAndOffset(node: Node, offset: number): NodeOffset {
     // html entities (eg. `&quot;`), however those entities get rendered in
     // the dom which throws off the offsets values, for example `&quot;`
     // becomes `"`, going from 6 characters to 1.
-    const encodedTextSubstring = originalText.substring(0, textOffset);
+    const encodedTextSubstring = originalText.substring(0, domOffset);
     const decodedText = decodeHTML(encodedTextSubstring);
-    const adjustment = textOffset - decodedText.length;
-    textOffset = Math.max(0, textOffset - adjustment);
+    const adjustment = domOffset - decodedText.length;
+    domOffset = Math.max(0, domOffset - adjustment);
+  }
+
+  // In certain cases (i.e. text length > 65536 chars), the DOM will break
+  // up text into multiple TextNodes. Loop through TextNodes, finding which
+  // contains the `domOffset`
+  const iterator = document.createNodeIterator(node, NodeFilter.SHOW_TEXT);
+  let textNode: Text, len: number;
+  do {
+    textNode = iterator.nextNode() as Text;
+  } while (textNode && (len = textNode.data.length) && domOffset > len && (domOffset -= len));
+
+  if (textNode === null) {
+    throw new Error(`Failed to find text node. Node: ${node.textContent}, offset: ${strOffset}`);
```

Comment:
is this message going to be displayed to the user or just logged internally? would be good to provide some i18n message text if its an end-user string

Reply:
Not end-user visible, so don't think it requires i18n. Will show in browser console, though; may help with debugging an issue if we can ask a user to look for errors/warnings there.